File: tests/test_kotlin/com/example/models/Order.kt
package com.example.models

import com.example.models.User

data class Order(
    val id: Long,
    val userId: String,
    val total: Double
) {
    fun getOwner(user: User): String {
        return user.name
    }

    fun formatTotal(): String {
        return String.format("$%.2f", total)
    }
}

File: tests/test_scala/com/example/models/Order.scala
package com.example.models

case class Order(
    id: Long,
    userId: String,
    total: Double
) {
    def getOwner(user: User): String = {
        user.name
    }

    def formatTotal: String = {
        f"$$${total}%.2f"
    }

    def isValid: Boolean = total > 0.0
}
